web_external/Datasets/ApplyMetadataView.js

```diff
@@ -0,0 +1,261 @@
+    //   file: MetadataFileModel
+    initialize: function (options) {
+        this.errors = options.errors;
+        this.file = options.file;
+        this.listenTo(this.errors, 'reset', this.render);
+        this.render();
+    },
+
+    render: function () {
+        this.$el.html(isic.templates.applyMetadataValidationPage({
+            errors: this.errors,
+            file: this.file
+        }));
+
+        return this;
+    }
+});
+
+// View to select a registered metadata file, validate the metadata, and save
+// the metadata to the dataset if validation is successful.
+isic.views.ApplyMetadataView = isic.View.extend({
+    events: {
+        'click #isic-apply-metadata-download-button': function () {
+            // Download selected metadata file
+            var fileModel = new girder.models.FileModel({_id: this.file.id});
+            fileModel.download();
+        },
+
+        'click #isic-apply-metadata-validate-button': function () {
+            var save = false;
+            this.validateMetadata(save);
+        }
+    },
+
+    initialize: function (settings) {
+        this.dataset = settings.dataset;
+
+        this.file = new isic.models.MetadataFileModel();
+        this.files = new isic.collections.MetadataFileCollection();
+
+        this.metadataErrorCollection = new isic.collections.MetadataErrorCollection();
+
+        this.selectFileView = new isic.views.ApplyMetadataSelectFileView({
+            collection: this.files,
+            parentView: this
+        });
+
+        this.validationView = new isic.views.ApplyMetadataValidationView({
+            errors: this.metadataErrorCollection,
+            file: this.file,
+            parentView: this
+        });
+
+        this.listenTo(this.selectFileView, 'changed', this.fileChanged);
+
+        this.listenTo(this.validationView, 'save', function () {
+            // Show confirmation dialog
+            girder.confirm({
+                text: '<h4>Really save metadata?</h4>',
+                escapedHtml: true,
+                yesText: 'Save',
+                yesClass: 'btn-primary',
+                confirmCallback: _.bind(function () {
+                    // Ensure dialog is hidden before continuing. Otherwise,
+                    // when validateMetadata() displays its modal alert dialog,
+                    // the Bootstrap-created element with class "modal-backdrop"
+                    // is erroneously not removed.
+                    $('#g-dialog-container').on('hidden.bs.modal', _.bind(function () {
+                        var save = true;
+                        this.validateMetadata(save);
+                    }, this));
+                }, this)
+            });
+        });
+
+        this.dataset.getRegisteredMetadata().done(_.bind(function (resp) {
+            this.files.reset(resp, {parse: true});
+        }, this));
+
+        this.render();
+    },
+
+    fileChanged: function (file) {
+        this.file.set(file.attributes);
```

Comment:
In general, I'd like to move away from this sort of thing, where we make copies of models, rather than more globally sharing a single instance. Of course, I see why you did this here, and it may indeed be the simplest way to handle it.

I wonder about the possibilities of either:
* Move the instantiation of `ApplyMetadataValidationView` into a separate method which gets called (effectively) within `filesChanged`. The `ApplyMetadataValidationView` would take a `file` as a parameter, then be destroyed and re-instantiated every time the `file` selection changes.
* Have `ApplyMetadataValidationView` expose a method to change its own file, re-rendering itself when that occurs.

I'd like to talk this through a bit further, just to be certain we're doing things the best way here.